lesson22/index.js

```diff
@@ -0,0 +1,242 @@
+    }
+
+    return fragment;
+};
+
+const createSortButtons = () => {
+    const ul = createElementWithClassName("ul", "sort-buttons");
+    ul.id = "js-sort-buttons";
+
+    const buttonConfig = [
+        { src: "../img/icon-both.svg", alt: "idを昇順に並び替える", dataSet: "default"},
+        { src: "../img/icon-asc.svg", alt: "idを降順に並び替える", dataSet: "asc"},
+        { src: "../img/icon-desc.svg", alt: "idを順不同に並び替える", dataSet: "desc"}
+    ]
+
+    const fragment = document.createDocumentFragment();
+    buttonConfig.forEach(item => {
+        const li = createElementWithClassName("li", "sort-buttons__item js-sort-button-item");
+        const button = createElementWithClassName("button", "sort-buttons__btn");
+        const img = createElementWithClassName("img", "sort-buttons__img");
+
+        li.dataset.buttonStatus = item.dataSet;
+        img.src = item.src;
+        img.alt = item.alt;
+        fragment.appendChild(li).appendChild(button).appendChild(img);
+    })
+    ul.appendChild(fragment);
+    return ul;
+};
+
+const renderSortButton = () => {
+    const sortTarget = [...document.querySelectorAll(".js-table-title")].filter(el => el.textContent === "ID" || el.textContent === "年齢");
+    
+    if(sortTarget) {
+        sortTarget.forEach(el => {
+            el.classList.add("is-target");
+            el.insertAdjacentElement('beforeend', createSortButtons());
+        })
+    }
+};
+
+const setButtonForInitDisplay = () => {
+    const defaultButtons = [...document.querySelectorAll('[data-button-status="default"]')];
+    defaultButtons.forEach(button => {
+        button.classList.add("is-active");
+    })
+}
+
+const addEventListenerForSortButtons = data => {
+    const sortButtons = [...document.querySelectorAll(".is-target")];
+
+    sortButtons.forEach(sortButton => {
+        sortButton.addEventListener("click", (e) => {
+            const activeColumn = e.currentTarget;
+
+            switchSortButtons(activeColumn);
+            
+            const trArray = [...document.querySelectorAll(".js-table-row")];
+            const sortData = createSortingData(data, activeColumn);
+    
+            trArray.forEach((tr, index) => {
+                const tdItems = tr.children;
+                changeTableContents(tdItems, sortData[index]);
+            })
+        });
+    })
+};
+
+const switchSortButtons = target => {
+    const activeButton = target.querySelector(".is-active");
+    const sortStatus = activeButton.dataset.buttonStatus;
+
+    activeButton.classList.remove("is-active");
+
+    switch (sortStatus) {
+        case "default":
+            target.querySelector("[data-button-status='asc']").classList.add("is-active");
+            break;
+        case "asc":
+            target.querySelector("[data-button-status='desc']").classList.add("is-active");
+            break;
+        default:
+            target.querySelector("[data-button-status='default']").classList.add("is-active");
+            break;
+    }
```

Comment:
IMO
It seems to me that it would be easier to read when the default is `default `, but what do you think?
```suggestion
 switch (sortStatus) {
    case "desc":
      target
        .querySelector("[data-button-status='default']")
        .classList.add("is-active");
      break;
    case "asc":
      target
        .querySelector("[data-button-status='desc']")
        .classList.add("is-active");
      break;
    default:
      target
        .querySelector("[data-button-status='asc']")
        .classList.add("is-active");
      break;
  }
```

Reply:
So true!
Thanks for letting us know!
Fixed.
[55c95b2](https://github.com/haru-programming/JavaScript-lessons/pull/27/commits/55c95b23df2a0cdd6bb3eff7ba23dbfbf41849ca)

Reply:
Thanks for the fix! Almost there.

It's even better to reference it from an object that's already defined!
Would it be possible for you to read my comment above again?

```js
 switch (sortStatus) {
    case Sort.Desc: //ココ
      target
        .querySelector("[data-button-status='default']")
        .classList.add("is-active");
      break;
    case Sort.Asc: //ココ
      target
        .querySelector("[data-button-status='desc']")
        .classList.add("is-active");
      break;
    default:
      target
        .querySelector("[data-button-status='asc']")
        .classList.add("is-active");
      break;
  }
```

Reply:
I am so sorry.....😭
I did not read it properly. Big mistake.
I apologize for your valuable time.
I fixed it!
[369b544](https://github.com/haru-programming/JavaScript-lessons/pull/27/commits/369b544072f23f1880162ce9d61f0e6f9034f9f3)